src/batch/src/task/context.rs

```diff
@@ -131,7 +99,8 @@ pub struct ComputeNodeContext {
     // to `last_mem_val` if new value comes in.
     cur_mem_val: Arc<AtomicUsize>,
 
-    stop_flag: Arc<StopFlag>,
+    shutdown_sender: Arc<Sender<ShutdownMsg>>,
```

Comment:
I think we should keep the sender in task execution rather than context. Context is usually used by executors.